components/director/internal/domain/formation/service.go

```diff
@@ -1002,106 +1070,126 @@ func (s *service) ResynchronizeFormationNotifications(ctx context.Context, forma
+			}
+			assignmentsForFormation, err := s.formationAssignmentService.GetAssignmentsForFormation(ctxWithTransact, tenantID, formationID)
+			if err != nil {
+				return errors.Wrapf(err, "while getting formation assignments for formation with ID %q", formationID)
+			}
+			for _, assignment := range assignmentsForFormation {
+				assignment.State = string(model.InitialAssignmentState)
+				formationassignment.ResetAssignmentConfigAndError(assignment) // reset the assignments
+				err = s.formationAssignmentService.Update(ctxWithTransact, assignment.ID, assignment)
+				if err != nil {
+					return err
+				}
 			}
 		}
-	}
 
-	resyncableFormationAssignments, err := s.formationAssignmentService.GetAssignmentsForFormationWithStates(faResyncTransactionCtx, tenantID, formationID,
-		[]string{string(model.InitialAssignmentState),
-			string(model.DeletingAssignmentState),
-			string(model.CreateErrorAssignmentState),
-			string(model.DeleteErrorAssignmentState)})
-	if err != nil {
-		return nil, errors.Wrap(err, "while getting formation assignments with synchronizing and error states")
-	}
-
-	failedDeleteErrorFormationAssignments := make([]*model.FormationAssignment, 0, len(resyncableFormationAssignments))
-	var errs *multierror.Error
-	for _, fa := range resyncableFormationAssignments {
-		operation := model.AssignFormation
-		if fa.State == string(model.DeleteErrorAssignmentState) || fa.State == string(model.DeletingAssignmentState) {
-			operation = model.UnassignFormation
-		}
-		var notificationForReverseFA *webhookclient.FormationAssignmentNotificationRequest
-		notificationForFA, err := s.formationAssignmentNotificationService.GenerateFormationAssignmentNotification(faResyncTransactionCtx, fa, operation)
+		resyncableFAs, err := s.formationAssignmentService.GetAssignmentsForFormationWithStates(ctxWithTransact, tenantID, formationID,
+			[]string{string(model.InitialAssignmentState),
+				string(model.DeletingAssignmentState),
+				string(model.CreateErrorAssignmentState),
+				string(model.DeleteErrorAssignmentState)})
 		if err != nil {
-			return nil, err
+			return errors.Wrap(err, "while getting formation assignments with synchronizing and error states")
 		}
 
-		reverseFA, err := s.formationAssignmentService.GetReverseBySourceAndTarget(faResyncTransactionCtx, fa.FormationID, fa.Source, fa.Target)
-		if err != nil && !apperrors.IsNotFoundError(err) {
-			return nil, err
+		for _, rfa := range resyncableFAs {
+			if rfa.State == string(model.DeleteErrorAssignmentState) {
+				failedDeleteErrorFormationAssignments = append(failedDeleteErrorFormationAssignments, rfa)
+			}
 		}
-		if reverseFA != nil {
-			notificationForReverseFA, err = s.formationAssignmentNotificationService.GenerateFormationAssignmentNotification(faResyncTransactionCtx, reverseFA, operation)
+		resyncableFormationAssignments = resyncableFAs
```

Comment:
It seems the declaration of resyncableFormationAssignments at the beggining of the method is not needed. We can remove it and use resyncableFAs, or use resyncableFormationAssignments on line 1101 instead of resyncableFAs

Reply:
The `resyncableFAs` is in the scope of the first transaction but the `resyncableFormationAssignments` is used outside of this scope, it' used in the second transaction. That's why we get the FAs in the first transaction and assigned it to `resyncableFormationAssignments` var so we can use it later.